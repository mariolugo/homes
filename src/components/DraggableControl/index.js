import React, { useState } from 'react';
import styled from 'styled-components';

const ControlStyled = styled.div`
  height: 35px;
  box-sizing: border-box;
  width: 290px;
  border-radius: 2px;
  background-color: rgb(255, 255, 255);
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;
  margin-bottom: 15px;
  cursor: pointer;
  position: absolute;
  bottom: 20px;
  left: 50%;
  margin-left: -148px;
  padding: 8px;
  span {
    font-size: 14px;
  }
`;

const HiddenCheckbox = styled.input.attrs({ type: 'checkbox' })`
  // Hide checkbox visually but remain accessible to screen readers.
  // Source: https://polished.js.org/docs/#hidevisually
  border: 0;
  clip: rect(0 0 0 0);
  clippath: inset(50%);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
`;

const Icon = styled.svg`
  fill: none;
  stroke: white;
  stroke-width: 2px;
`;

const StyledCheckbox = styled.div`
  display: inline-block;
  width: 16px;
  height: 16px;
  border: 1px solid black;
  background: ${(props) => (props.checked ? 'rgb(141, 122, 255)' : 'white')};
  border-radius: 3px;
  transition: all 150ms;
  margin-right: 10px;
  ${HiddenCheckbox}:focus + & {
    box-shadow: 0 0 0 3px rgb(141, 122, 255);
  }
  ${Icon} {
    visibility: ${(props) => (props.checked ? 'visible' : 'hidden')};
  }
`;

const CheckboxContainer = styled.div`
  display: inline-block;
  vertical-align: middle;
`;

/**
 * Component used for the map, in here we have a checkbox and we can add logic to handle the checkbox.
 * For example we can activate the checkbox and filter the homes if the map changes
 */
const DraggableControl = () => {
  const [active, setActive] = useState(false);

  const onCheckboxChange = () => {
    setActive((prev) => !prev);
  };

  return (
    <ControlStyled>
      <div>
        <form>
          <label htmlFor="checkbox">
            <CheckboxContainer>
              <HiddenCheckbox id="checkbox" checked={active} onChange={onCheckboxChange} />
              <StyledCheckbox checked={active}>
                <Icon viewBox="0 0 24 24">
                  <polyline points="20 6 9 17 4 12" />
                </Icon>
              </StyledCheckbox>
            </CheckboxContainer>
            <span>Filtrar mientras te mueves en el mapa</span>
          </label>
        </form>
      </div>
    </ControlStyled>
  );
};

export default DraggableControl;
