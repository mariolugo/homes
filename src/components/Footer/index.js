import React from 'react';
import styled from 'styled-components';
import { Col, Row } from 'react-styled-flexboxgrid';
import { WrapperGrid } from '../';
import FooterLogo from '../../assets/icons/footer-logo.svg';
import Messenger from '../../assets/icons/messenger.svg';

const FooterStyled = styled.div`
  background-color: white;
  border-top: 1px solid ${({ theme }) => theme.colors.grayLight};
  margin: 0px auto;
  min-height: 60px;

  @media only screen and (max-width: 600px) {
    padding-top: 10px;
  }
`;

const LinkStyled = styled.p`
  font-size: 13px;
`;

const FooterLink = styled(LinkStyled)`
  text-align: center;
`;

const ListStyled = styled.ul`
  list-style: none;
  padding-left: 0px;
`;

const ListItemStyled = styled.li`
  svg {
    display: inline-block;
    margin-right: 5px;
    position: relative;
    top: 2px;
  }
  p {
    display: inline-block;
    margin: 0;
  }

  @media only screen and (max-width: 600px) {
    text-align: center;
  }
`;

const LogoContainer = styled(Col)`
  display: flex;
  padding-right: 0;
  align-items: center;
  justify-content: center;
`;

const ColCentered = styled(Col)`
  display: flex;
  align-items: center;
  justify-content: center;
`;

/**
 * Main footer of the application
 */
const Footer = () => (
  <FooterStyled>
    <WrapperGrid>
      <Row>
        <Col xs={12} sm={4}>
          <Row>
            <LogoContainer xs={12} sm={3}>
              <FooterLogo />
            </LogoContainer>
            <Col xs={12} sm={9}>
              <ListStyled>
                <ListItemStyled>
                  <LinkStyled>hey@homie.mx</LinkStyled>
                </ListItemStyled>
                <ListItemStyled>
                  <LinkStyled>5015 1546</LinkStyled>
                </ListItemStyled>
                <ListItemStyled>
                  <Messenger />
                  <LinkStyled>Escríbenos por facebook</LinkStyled>
                </ListItemStyled>
              </ListStyled>
            </Col>
          </Row>
        </Col>
        <Col xs={12} sm={3}>
          <ListStyled>
            <ListItemStyled>
              <LinkStyled>Te buscamos depa</LinkStyled>
            </ListItemStyled>
            <ListItemStyled>
              <LinkStyled>Blog</LinkStyled>
            </ListItemStyled>
            <ListItemStyled>
              <LinkStyled>Equipo</LinkStyled>
            </ListItemStyled>
          </ListStyled>
        </Col>
        <ColCentered xs={12} sm={2}>
          <FooterLink>Aviso de privacidad</FooterLink>
        </ColCentered>
        <ColCentered xs={12} sm={3}>
          <FooterLink>Términos y condiciones</FooterLink>
        </ColCentered>
      </Row>
    </WrapperGrid>
  </FooterStyled>
);

export default Footer;
