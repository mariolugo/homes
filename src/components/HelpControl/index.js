import React from 'react';
import styled from 'styled-components';
import Image from 'next/image';

const ControlStyled = styled.div`
  width: 50px;
  height: 50px;
  background-color: white;
  cursor: pointer;
  position: absolute;
  bottom: 20px;
  right: 50%;
  margin-right: 10px;
  box-sizing: border-box;
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;
  padding: 13px 9px;
  border-radius: 2px;
`;

/**
 * Help control that is required in the map.
 */
const HelpControl = () => {
  return (
    <ControlStyled>
      <Image src="/images/help_map.png" alt="Help Map" width={35} height={35} />
    </ControlStyled>
  );
};

export default HelpControl;
