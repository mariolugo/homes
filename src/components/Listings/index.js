import React from 'react';
import styled from 'styled-components';
import { Col, Row } from 'react-styled-flexboxgrid';
import { Post } from '../';

const ListingsContainerStyled = styled.div``;

const ListingsContainer = ({ posts = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15] }) => (
  <ListingsContainerStyled>
    <Row>
      {posts.length > 0 &&
        posts.map((p) => (
          <Col key={p} xs={12} md={6}>
            <Post />
          </Col>
        ))}
    </Row>
  </ListingsContainerStyled>
);

export default ListingsContainer;
