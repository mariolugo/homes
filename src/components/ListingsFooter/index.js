import React from 'react';
import styled from 'styled-components';

const ListingsFooterStyled = styled.div`
  p {
    color: ${({ theme }) => theme.colors.textPrimary};
  }
`;

const Title = styled.p`
  text-align: center;
  font-family: 'Open Sans Bold', sans-serif;
`;

const Subtitle = styled.p`
  text-align: center;
`;

const Button = styled.button`
  border-radius: 2px;
  border: none;
  width: 300px;
  display: block;
  height: 35px;
  background-color: ${({ theme }) => theme.colors.secondary};
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;
  color: white;
  font-family: 'Open Sans Bold', sans-serif;
  line-height: 19px;
  padding: 8px 0px;
  margin: 20px auto;
`;

const ListingsFooter = () => (
  <ListingsFooterStyled>
    <Title>¿No encuentras inmueble?</Title>
    <Subtitle>Dinos qué quieres y nosotros te lo buscamos</Subtitle>
    <Button>Encuéntrenme un inmueble</Button>
  </ListingsFooterStyled>
);

export default ListingsFooter;
