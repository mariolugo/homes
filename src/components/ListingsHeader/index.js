import React from 'react';
import styled from 'styled-components';
import { Col, Row } from 'react-styled-flexboxgrid';
import Filter from '../../assets/icons/filter.svg';

const ListingHeaderStyled = styled.div`
  width: 100%;
  margin-bottom: 15px;
`;

const FilterButton = styled.button`
  height: 50px;
  background-color: ${({ theme }) => theme.colors.primary};
  border: none;
  border-radius: 3px;
  color: white;
  width: 100%;
  cursor: pointer;
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;

  svg {
    display: inline-block;
    margin-right: 5px;
    position: relative;
    top: 2px;
  }

  p {
    display: inline-block;
    font-family: 'Open Sans Bold', sans-serif;
    font-size: 16px;
    margin: 0;
  }

  @media only screen and (max-width: 500px) {
    margin-top: 15px;
  }
`;

const InfoBox = styled.div`
  background-color: white;
  height: 50px;
  max-height: 50px;
  border-radius: 3px;
  box-shadow: rgba(0, 0, 0, 0.5) 0px 1px 2px 0px;

  h3 {
    font-size: 16px;
    line-height: 1;
    padding: 7px 0px 0px;
    margin: 0px 0px 0px 10px;
    color: ${({ theme }) => theme.colors.primary};
    font-family: 'Open Sans Bold', sans-serif;
  }

  h4 {
    font-size: 14px;
    line-height: 1;
    padding: 7px 0px 0px;
    margin: 0px 0px 0px 10px;
    color: ${({ theme }) => theme.colors.textPrimary};
  }
`;

const ListingHeaders = () => (
  <ListingHeaderStyled>
    <Row>
      <Col xs={12} sm={9}>
        <InfoBox>
          <h3>Ciudad de Mexico</h3>
          <h4>230 departamentos</h4>
        </InfoBox>
      </Col>
      <Col xs={12} sm={3}>
        <FilterButton>
          <Filter />
          <p>Filtros</p>
        </FilterButton>
      </Col>
    </Row>
  </ListingHeaderStyled>
);

export default ListingHeaders;
