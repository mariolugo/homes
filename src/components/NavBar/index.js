import React from 'react';
import styled from 'styled-components';
import { Col, Row } from 'react-styled-flexboxgrid';
import { WrapperGrid, Logo } from '../';
import Favorites from '../../assets/icons/favorites.svg';
import { useDeviceDetect } from '../../utils';

const NavBarStyled = styled.nav`
  background-color: white;
  border-bottom: 1px solid ${({ theme }) => theme.colors.pink};
  position: fixed;
  width: 100%;
  margin: 0px auto;
  min-height: 60px;
  p {
    color: ${({ theme }) => theme.colors.textPrimary};
  }
`;

const RowStyled = styled(Row)`
  min-height: 60px;
`;

const ColStyled = styled(Col)`
  min-height: 60px;
  display: flex;
  align-items: center;
`;

const ColStyledMobile = styled(ColStyled)`
  min-height: 60px;
  display: flex;
  align-items: center;
  @media only screen and (max-width: 640px) {
    padding-left: 30px;
  }
`;

const MenuLinksStyled = styled(Col)`
  display: flex;
  justify-content: flex-end;
`;

const FavoritesIcon = styled(Favorites)`
  margin-right: 5px;
  position: relative;
  top: 2px;
`;

const LinkStyled = styled.p`
  font-size: 13px;
`;

const ListStyled = styled.ul`
  list-style: none;
  margin: 0;
  height: 100%;
  display: table;

  @media only screen and (max-width: 640px) {
    padding-left: 0px;
  }
`;

const ListItemStyled = styled.li`
  display: table-cell;
  padding: 0px 15px;
  height: 100%;
  vertical-align: middle;
  transition: background-color 0.2s linear 0s;
  &:hover {
    background-color: ${({ theme }) => theme.colors.linkHover};
  }

  @media only screen and (max-width: 640px) {
    padding-left: 0px;
  }
`;

const links = [
  {
    id: 1,
    text: '¿Cómo funciona?',
    onMobile: false,
  },
  {
    id: 2,
    text: 'Soy propietario',
    onMobile: false,
  },
  {
    id: 3,
    text: 'Iniciar sesión',
    onMobile: true,
  },
];

const NavBar = () => {
  const { isMobile } = useDeviceDetect();
  return (
    <NavBarStyled>
      <WrapperGrid>
        <RowStyled>
          <ColStyledMobile xs={5} md={2}>
            <Logo />
          </ColStyledMobile>
          <ColStyled xs={3} md={4}>
            <FavoritesIcon />
            <LinkStyled>Favoritos</LinkStyled>
          </ColStyled>
          <MenuLinksStyled xs={4} md={6}>
            <ListStyled>
              {links.map((l) => {
                if (isMobile && !l.onMobile) return null;
                return (
                  <ListItemStyled key={l.id}>
                    <LinkStyled>{l.text}</LinkStyled>
                  </ListItemStyled>
                );
              })}
            </ListStyled>
          </MenuLinksStyled>
        </RowStyled>
      </WrapperGrid>
    </NavBarStyled>
  );
};

export default NavBar;
