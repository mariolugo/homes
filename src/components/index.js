import NavBar from './NavBar';
import Listings from './Listings';
import Map from './Map';
import Footer from './Footer';
import Post from './Post';
import WrapperGrid from './WrapperGrid';

export { NavBar, Listings, Map, Footer, Post, WrapperGrid };
