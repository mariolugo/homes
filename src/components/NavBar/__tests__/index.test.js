import { mountWithTheme } from '../../../../tests/themeProvider';
import { NavBar } from '../../../components';

/** @test {NavBar Component} */
describe('NavBar Component', () => {
  it('should render without crashing', () => {
    const wrapper = mountWithTheme(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
