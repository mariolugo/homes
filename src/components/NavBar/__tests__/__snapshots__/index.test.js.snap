// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component should render without crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-hKgILt feuioQ"
      >
        <div
          class="sc-bdfBwQ sc-kLgntA guSBMN iXAtuJ"
        >
          <div
            class="sc-gsTCUz sc-eCssSg ikQfAY cdLCTx"
          >
            <div
              class="sc-dlfnbm sc-jSgupP sc-gKsewC enWnyg kGOcbo dEYPlJ"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;charset=utf-8,<svg width=\\"94\\" height=\\"32\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                    style="max-width: 100%; display: block;"
                  />
                </div>
                <img
                  alt="Logo"
                  decoding="async"
                  src="/_next/image?url=%2Fimages%2Flogo.png&w=320&q=75"
                  srcset="/_next/image?url=%2Fimages%2Flogo.png&w=320&q=75 1x"
                  style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
            <div
              class="sc-dlfnbm sc-jSgupP cHhzLc kGOcbo"
            >
              <svg
                class="sc-fubCfw ExwbA"
                height="14"
                width="15"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M13.84 2.016A4.099 4.099 0 0 0 11.037.933a4.099 4.099 0 0 0-2.802 1.083l-.736.687-.736-.687A4.099 4.099 0 0 0 3.962.933 4.099 4.099 0 0 0 1.16 2.016c-1.547 1.445-1.547 3.786 0 5.23l6.34 5.917 6.34-5.917c1.547-1.444 1.547-3.785 0-5.23"
                  fill="#FF1555"
                />
              </svg>
              <p
                class="sc-pFZIQ evzlSA"
              >
                Favoritos
              </p>
            </div>
            <div
              class="sc-dlfnbm sc-iBPRYJ ktyqAo bttmjO"
            >
              <ul
                class="sc-jrAGrp crlQdU"
              >
                <li
                  class="sc-kEjbxe gsuFIM"
                >
                  <p
                    class="sc-pFZIQ evzlSA"
                  >
                    ¿Cómo funciona?
                  </p>
                </li>
                <li
                  class="sc-kEjbxe gsuFIM"
                >
                  <p
                    class="sc-pFZIQ evzlSA"
                  >
                    Soy propietario
                  </p>
                </li>
                <li
                  class="sc-kEjbxe gsuFIM"
                >
                  <p
                    class="sc-pFZIQ evzlSA"
                  >
                    Iniciar sesión
                  </p>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-hKgILt feuioQ"
    >
      <div
        class="sc-bdfBwQ sc-kLgntA guSBMN iXAtuJ"
      >
        <div
          class="sc-gsTCUz sc-eCssSg ikQfAY cdLCTx"
        >
          <div
            class="sc-dlfnbm sc-jSgupP sc-gKsewC enWnyg kGOcbo dEYPlJ"
          >
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;charset=utf-8,<svg width=\\"94\\" height=\\"32\\" xmlns=\\"http://www.w3.org/2000/svg\\" version=\\"1.1\\"/>"
                  style="max-width: 100%; display: block;"
                />
              </div>
              <img
                alt="Logo"
                decoding="async"
                src="/_next/image?url=%2Fimages%2Flogo.png&w=320&q=75"
                srcset="/_next/image?url=%2Fimages%2Flogo.png&w=320&q=75 1x"
                style="visibility: visible; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
          <div
            class="sc-dlfnbm sc-jSgupP cHhzLc kGOcbo"
          >
            <svg
              class="sc-fubCfw ExwbA"
              height="14"
              width="15"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M13.84 2.016A4.099 4.099 0 0 0 11.037.933a4.099 4.099 0 0 0-2.802 1.083l-.736.687-.736-.687A4.099 4.099 0 0 0 3.962.933 4.099 4.099 0 0 0 1.16 2.016c-1.547 1.445-1.547 3.786 0 5.23l6.34 5.917 6.34-5.917c1.547-1.444 1.547-3.785 0-5.23"
                fill="#FF1555"
              />
            </svg>
            <p
              class="sc-pFZIQ evzlSA"
            >
              Favoritos
            </p>
          </div>
          <div
            class="sc-dlfnbm sc-iBPRYJ ktyqAo bttmjO"
          >
            <ul
              class="sc-jrAGrp crlQdU"
            >
              <li
                class="sc-kEjbxe gsuFIM"
              >
                <p
                  class="sc-pFZIQ evzlSA"
                >
                  ¿Cómo funciona?
                </p>
              </li>
              <li
                class="sc-kEjbxe gsuFIM"
              >
                <p
                  class="sc-pFZIQ evzlSA"
                >
                  Soy propietario
                </p>
              </li>
              <li
                class="sc-kEjbxe gsuFIM"
              >
                <p
                  class="sc-pFZIQ evzlSA"
                >
                  Iniciar sesión
                </p>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
